src/test/java/picard/pedigree/SexTest.java

```diff
@@ -0,0 +1,54 @@
+package picard.pedigree;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.PicardException;
+
+import java.util.Arrays;
+
+public class SexTest {
+    @DataProvider(name="testSexDataProviderInt")
+    public Object[][] testSexDataProviderInt() {
+        return new Object[][] {
+                new Object[]{Sex.Male, 1, true},
+                new Object[]{Sex.Female, 2, true},
+                new Object[]{Sex.Unknown, 0, true},
+                new Object[]{Sex.NotReported, 0, false},
+                new Object[]{Sex.Female, 1, false},
+                new Object[]{Sex.Unknown, 2, false},
+                };
+    }
+
+    @DataProvider(name="testSexDataProviderStr")
+    public Object[][] testSexDataProviderStr() {
+        return new Object[][] {
+                new Object[]{Sex.Male, "M", true},
+                new Object[]{Sex.Female, "Female", true},
+                new Object[]{Sex.Unknown, "Unknown", true},
+                new Object[]{Sex.NotReported, "NotReported", true},
+                new Object[]{Sex.Female, "M", false},
+                new Object[]{Sex.Unknown, "female", false}
+        };
+    }
+
+    /**
+     *Test Methods within Sex enum
+     */
+
+    @Test(dataProvider="testSexDataProviderInt")
+    public void testFromCode(final Sex sex, final int i, final boolean b){
```

Comment:
I think it would probably be simpler to just take as input the integer and the expected Sex, and the test that Sex.fromCode(i) == sex.  You don't really need the false tests, once you know that Sex.fromCode returns one value, you know it doesn't return any of the other values.